ipatests/test_integration/test_sssd.py

```diff
@@ -403,3 +406,193 @@ def test_trustdomain_disable_disables_subdomain(self):
                 assert 'no such user' in res.stderr_text
         # verify the user can be retrieved after re-enabling trustdomain
         self.master.run_command(['id', user])
+
+
+class TestNestedMembers(IntegrationTest):
+    num_clients = 1
+
+    @classmethod
+    def install(cls, mh):
+        tasks.install_master(cls.master)
+        tasks.install_client(cls.master, cls.clients[0])
+
+    def test_nested_group_members(self):
+        """Nested group memberships should be honoured
+
+        "c" should be a child of "b"
+        so that parent child relationship is as follows:
+        "a"->"b"->"c"
+
+        testuser001 is direct member of "a" and as a result
+        member of "b" and "c"".
+        Now if one adds a direct membership to "b"
+        nothing will change.
+
+        Now if one removes the direct membership to "b"
+        nothing should change, the memberships should be honored
+        Linked Issue: https://pagure.io/SSSD/sssd/issue/3636
+        """
+        master = self.master
+        client = self.clients[0]
+
+        username = "testuser001"
+        userpasswd = 'Secret123'
+
+        # tmpdir for creating ssh keys for user
+        # since ssh does not work well paramiko
+        tmpdir = tempfile.mkdtemp(prefix="tmp-")
+
+        # add a user and set password
+        tasks.create_active_user(master, username, userpasswd)
+        tasks.kinit_admin(master)
+
+        pub_keys = []
+
+        ssh_key_pair = tasks.generate_ssh_keypair()
+        pub_keys.append(ssh_key_pair[1])
+        with open(os.path.join(
+                tmpdir, 'ssh_priv_key'), 'w') as fp:
+            fp.write(ssh_key_pair[0])
+
+        keys_opts = ' '.join(['--ssh "{}"'.format(k) for k in pub_keys])
```

Comment:
Don't construct commands as string. Always use a list, please.

Reply:
modified the command to use a list instead of strings